### DISCUSSION TITLE
Openbor questions

### DISCUSSION THREAD

**Discussion by tooonyo** [OnionUI/Onion#519] [Q&A]
Hi ! I'm new to Miyoo mini but the first thing I did was installing OnionOS :-D
I'm very excited to play Openbor. I saw there was an app already installed in the "App" folder of the sd card, but it can't play a lot of PAK files... Maybe it's an old version ? 

I saw in the onion installer an experimental core for BOR. I activated it, put the paks in /RApp/OpenBOR/paks but I don't see where is the openbor app. Is there a problem ?

And finally, a friend of mine gave me a link (https://github.com/steward-fu/miyoo-mini/releases) with a lot of apps that we can install on the miyoo. There are four versions of openbor (the most recent is 20220414), but none of them is working... It makes the handheld reboot infinitely.

In conclusion, can you help me to have a recent version of openbor please ? What are your advices ?

Thanks in advance

**Comment by dwmccoy** [comment]
It is my opinion that the next release of Onion OS will finally have a fix for the OpenBOR issues.  I had posted a workaround in the meantime.  That information can be found here #185.

**Reply by tooonyo** [reply]
I am on 3.10.6 and openbor only works in the APP folder, but I think it's an old version that was already on the SD card whel I received the Miyoo. I tried the method in the #185 issue : Openbor appears on the consoles list but when I launch a game, it's stuck on the miyoo loading screen. If I try to put the April 22nd version of OpenBOR in the APP folder, the Miyoo is in an infinite reboot circle. 

Do you have any clue ? Thanks !

**Reply by dwmccoy** [reply]
In my solution, you CAN NOT install OpenBOR from the OnionOS installer.  If did Install it from Onion OS, Un-Install it.  You CAN NOT place the 04-22-2022 version of OpenBOR in the APP folder.  Your in a constant reboot because you have placed OpenBor version 04-22-2002 in the APP folder.  I really can't detail the instructions I outlined in #185 much more.  Please re-read the guide.  If your not able to make it work, I guess you'll have to wait and see if they fix it in the next version of Onion OS.

**Reply by tooonyo** [reply]
Thanks ! I guess I'll wait because I didn't install the 20220422 from the onion installer and I followed exactly the instructions of the #185 
Openbor is in the EMU folder and the files are modified as explained.

**Comment by Sichroteph** [comment]
I used the config files proposed, I'll also check an alternative build that is promising in the following days.

**Comment by schmurtzm** [comment]
Still not working, on the last Onion 3.11.2 Openbor UI is OK but running a .pak rom is not OK (sound OK, but display is strange, see screenshot) : 
![image](https://user-images.githubusercontent.com/7110113/172190416-2a096649-71b6-428d-8ac1-da884d7a9713.png)

I have successfuly run a rom with the advices from here : 
https://github.com/jimgraygit/Onion/issues/124#issuecomment-1130203720
But this is an external compilation from Onion.


Btw, Openbor could be added in the new list of PORTS too ;)

**Reply by dwmccoy** [reply]
I am no expert in this but this looks like a video refresh rate issue.  I know that some OpenBOR games work without issue using the version included with Onion OS 3.11.2.  My best guess is maybe different games access the video hardware/memory/driver differently causing the video display issue.... I know that my solution posted in #185 still works with Onion OS 3.11.2 and the video issue does not occur.

**Comment by tooonyo** [comment]
I followed the #124 advice : NOK (roms in the listing but infinite loading screen, not even openbor interface. I tried paks that have already worked with other version of openbor)
I installed the experimental core form Onion Installer : NOK (nothing happens)
The only way to make it work was to put the original openbor of the miyoo stock sd in the app folder, but it was a very old version I assume

Do you think there's a solution to make it work correctly ? I don't think it's only a video display issue because I managed to make the paks run with the stock sd.

Thanks !

**Comment by dwmccoy** [comment]
I have tested several PAK files that have the same video issue shown in this thread when using the OpenBOR that came with the Stock SDCARD.  I also have the same video issue with several PAK files using the version of OpenBOR included in Onion OS 3.11.2.  The difference in the OpenBOR version 04-22-2002 that works for me as outlined in #185 is that the 04-22-2002 version uses libGL instead of libSDL and also launches the PAK file without the need of the OpenBOR interface.  libGL and libSDL are two totally different video display engines.  I know the stock version and the Onion version are both based on SDL 1.x.  I don't know if it can be compiled against the libSDL 2.x or not.  It is my opinion that the PAK file is making a video request that the libSDL 1.x library can't perform causing the video anomaly.  I have tested my solution on Onion OS 3.11.2 and it works.  I did NOT follow the solution from #124 nor did I follow the guide that was on Reddit.  The solution I outlined #185 requires the 04-22-2002 version and it needs to be placed in the /EMU/OpenBOR folder.  Any attempt to put it in /APP or /RApp will cause the mini to go into a reboot loop.
